src/redux/books.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-unused-vars */
+import { createSlice, createAction } from '@reduxjs/toolkit';
+
+// Action
+export const apiCallBegan = createAction('api/callBegan');
+export const apiCallSuccess = createAction('api/callSuccess');
+export const apiCallFailed = createAction('api/callFailed');
+
+// Entity Slice
+const slice = createSlice({
+  name: 'books',
+  initialState: {
+    list: [],
+    loading: false,
+  },
+  reducers: {
+    booksRequested: (books, action) => {
+      books.loading = true;
+    },
+    booksRequestFailed: (books, action) => {
+      books.loading = false;
+    },
+    booksReceived: (books, action) => {
+      books.list = action.payload;
+      books.loading = false;
+    },
+  },
+});
+
+export const {
+  bookAdded, bookRemoved, booksReceived, booksRequested, booksRequestFailed,
+} = slice.actions;
+
+// action creators
+const url = '/books';
+export const loadBooks = () => apiCallBegan({
+  url,
+  method: 'GET',
+  onStart: booksRequested.type,
+  onSuccess: booksReceived.type,
+  onError: booksRequestFailed.type,
+});
+
+export const addBook = (book) => apiCallBegan({
+  url,
+  method: 'POST',
+  body: JSON.stringify(book),
+  onStart: booksRequested.type,
+  onSuccess: booksReceived.type,
+  onError: booksRequestFailed.type,
+});
+
+export const removeBook = (id) => apiCallBegan({
+  url: `${url}/${id}`,
+  method: 'DELETE',
+  body: JSON.stringify({ item_id: id }),
+  onStart: booksRequested.type,
+  onSuccess: booksReceived.type,
+  onError: booksRequestFailed.type,
+});
+
+export default slice.reducer;
```

Comment:
- [ ] It's the first time I see this approach. I love it. Only one issue I want to address. In Redux, we use actions to handle the state of the books array. The state books should be updated each time I add a book or remove a book. It means the book should be rendered in the layout instantly after adding it and it should be removed from the layouts after clicking on the remove button. Let's work on that.